docs/user/api.asciidoc

```diff
@@ -5,6 +5,8 @@ Some {kib} features are provided via a REST API, which is ideal for creating an
 integration with {kib}, or automating certain aspects of configuring and
 deploying {kib}.
 
+NOTE: Access to internal {kib} API endpoints will be restricted in 9.0. For more information, refer to https://github.com/elastic/kibana/issues/163654.
```

Comment:
I went with a note rather than a warning because, strictly speaking, end users should only be consuming public APIs.

Reply:
NOTE makes sense to me 👍 

Is it useful to link to the github issue? I'm not sure that it would help a user to read through that issue. We could ask the docs team for input, but maybe it's sufficient to just say something like:

NOTE: Access to internal {kib} API endpoints will be restricted in 9.0. Please move any integrations to publicly documented APIs.

Reply:
I agree that providing advice here rather than sending them to another page is better.

Reply:
++ we're not intending to document the `restrictInternalAPIs` setting and it makes sense to give advice instead of the issue.

Reply:
Done in [`db31e15`](https://github.com/elastic/kibana/pull/191943/commits/db31e15d41be2a9c99cb0998195a703f087443ac)
![Screenshot 2024-09-03 at 10 21 28](https://github.com/user-attachments/assets/751d4f15-21ce-473f-bd43-d2efb6434e3a)